python/plugins/processing/algs/gdal/gdalcalc.py

```diff
@@ -138,6 +140,12 @@ def initAlgorithm(self, config=None):
                 type=QgsProcessingParameterNumber.Double,
                 defaultValue=None,
                 optional=True))
+
+        if GdalUtils.version() >= 3030000:
+            self.addParameter(QgsProcessingParameterExtent(self.EXTENT,
+                                                           self.tr('Output extent'),
+                                                           optional=True))
```

Comment:
@nirvn Can you `setHelp` the parameter, please? Now that there's a dedicated function, we should document every single (new) parameter. It would help users directly in the GUI and would avoid documenters to try/guess or browse a third party docs to find out how it works. Thanks.

Reply:
@DelazJ , I don't mind doing it, loving the effort around better documenting algorithms here. That said, in this case, I'm not sure what the help would do that the description doesn't :)

Desc: output extent
Help: this parameter provides a mean to set the extent of the output

? :)

Reply:
I first thought you were adding [extent](https://gdal.org/programs/gdal_calc.html#cmdoption-extent) but looks like it's [projwin](https://gdal.org/programs/gdal_calc.html#cmdoption-projwin). I'd say, the help can be a reply to 'in what this seems to be quite "very useful" to you'?
Doing it the lazy way, it can be `Custom extent of the output raster` but a point [the docs ](https://docs.qgis.org/testing/en/docs/user_manual/processing_algs/gdal/rastermiscellaneous.html#raster-calculator) does not address is how CRS are managed with this alg. So should the output extent be specified in the first layer CRS or anyone is good enough? Is it a crop? a reprojection? Just few questions I could ask myself when trying to describe such a feature with provided info but without your background.
While at it, what do you think about reducing the calc parameter's label and move details in a help?